test/plugins/features/movies/controller.test.js

```diff
@@ -23,4 +29,81 @@ describe('movie controller', () => {
 
   });
 
+  describe('findAll', () => {
+
+    beforeEach(() => {
+      return Knex('movies').insert([movieOne, movieTwo]);
+    });
+
+    it('finds all movies', () => {
+      const query = {};
+
+      return Controller.findAll(query)
+      .then((movies) => {
+        expect(movies).to.have.lengthOf(2);
+      });
+    });
+
+    it('finds movies matching title', () => {
+      const query = { title: movieOne.title };
+
+      return Controller.findAll(query)
+      .then((movies) => {
+        expect(movies.models[0].get('title')).to.eql(query.title);
+      });
+    });
+
+    it('finds movies released in release_year', () => {
+      const query = { release_year: { eq: movieTwo.release_year } };
+
+      return Controller.findAll(query)
+      .then((movies) => {
+        expect(movies.models[0].get('release_year')).to.eql(query.release_year.eq);
+      });
+    });
+
+    it('finds movies released before release_year', () => {
+      const query = { release_year: { lt: movieOne.release_year + 1 } };
+
+      return Controller.findAll(query)
+      .then((movies) => {
+        expect(movies.models[0].get('release_year')).to.be.below(query.release_year.lt);
+      });
+    });
+
+    it('finds movies released after release_year', () => {
+      const query = { release_year: { gt: movieTwo.release_year - 1 } };
+
+      return Controller.findAll(query)
+      .then((movies) => {
+        expect(movies.models[0].get('release_year')).to.be.above(query.release_year.gt);
+      });
+    });
+  });
+
+  describe('findById', () => {
+
+    beforeEach(() => {
+      return Knex('movies').insert([movieOne, movieTwo]);
+    })
+
+    it('finds movie by id', () => {
+      const params = { id: 1 };
+
+      return Controller.findById(params.id)
+      .then((movie) => {
+        expect(movie.id).to.eql(params.id);
+      });
+    });
+
+    it('returns nothing if no movie has given id', () => {
+      const params = { id: 0 };
+
+      return Controller.findById(params.id)
+      .then((movie) => {
+        expect(movie).to.not.exist;
+      });
```

Comment:
I'm not sure which is better, to return nothing or to throw an error. What do you think?